src/Products/ZCatalog/query.py

```diff
@@ -65,7 +68,7 @@ def __init__(self, request, iid, options=(), operators=('or', 'and'),
         param = request[iid]
         keys = None
 
-        if isinstance(param, dict):
+        if isinstance(param, dict) or isinstance(param, record):
```

Comment:
I think this should work as well:
```suggestion
        if isinstance(param, (dict, record)):
```

Reply:
Thanks @ale-rt ! I made your suggestion.